pyccel/ast/sysext.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+""" Module containing objects from the sys module understood by pyccel
+"""
+from .core import PyccelFunctionDef, Module
+from .internals import PyccelInternalFunction
+from .datatypes import NativeVoid
+from .internals import LiteralInteger
+from .operators import PyccelUnarySub
+
+class SysExit(PyccelInternalFunction):
+    """Represents a call to  sys.exit
+
+    arg : LiteralInteger, PyccelUnarySub(LiteralInteger)
```

Comment:
as in [python docs for sys.exit](https://docs.python.org/3/library/sys.html#sys.exit) there are two stats:
- Success exit: we may have `exit( )` or `exit(0)`
- Error exit: `exit(non-zero-int)` or `exit(msg-error)`
I wonder if you can add the support for the case of empty arg and msg error?

Reply:
I already added the support for empty calls but with the exit(object) we don't have support for printing to the standard error

Reply:
> I couldn't figure out how to test the exit error from Pytest I did try to use [this](https://stackoverflow.com/questions/62427205/in-python-3-using-pytest-how-do-we-test-for-exit-code-exit1-and-exit0-fo) but it doesn't catch the exit for the compiled version and the test just exits

we can fairly see that the python exit function and C exit function have different behavior, I am confused about these two options: keep it simple like what you did, or use `PyExc_SystemExit` from python C Api.
@EmilyBourne what do you think?

Reply:
hmm, maybe we can use our c low-level function `write`, for writing to stderr, we will need just to import the libraries required to do that.
what do you think?

Reply:
I don't think it's a good idea to use write as it's hard to format the output.
we can use `fprintf` for that and use the same already written _print_PythonPrint to generate the code for it.
but I couldn't figure out how to write to `stderr` on fortran, need someone familiar with it to add that

Reply:
https://stackoverflow.com/a/8508757/5957195